### ISSUE TITLE
GroupedList's section list isn't able to set an ariaLabel="<[parent header text>"

### ISSUE THREAD

**Issue by chihch** [microsoft/fluentui#12930]
### Environment Information

- **Package version(s)**: 7.94.0
- **Browser and OS versions**: Windows 10

### Describe the issue:

Using a GroupedList with different section/subList, each section root (the one with role="grid") is not able to set the ariaLabel="section header text"

if you see the codepen and inspect the element tree, you will see that each section of GroupedList (i.e. role="grid") doesn't get the ariaLabel, but it's parent (i.e. the section header gets that). Furthermore, if you use a custom header (i.e. pass in onRenderHeader), this ariaLabel doesn't even show up on that section header.

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/chihch/pen/vYNJeYR

#### Actual behavior:

The property (aka. ariaLabel) that can be configured/form in IGroup for each group and set to 'group' property of GroupedList only applies to the section header (i.e. the section header gets the ariaLabel)

#### Expected behavior:

each section of GroupedList can have an ariaLabel="section header text", or have ariaLabelledby that points to it's header

**Comment by aneeshack4** [comment]
I ran the Accessibility Insights tool on this and the ARIA label not being set on the section list doesn't get flagged as an error which makes me think that is the expected behavior, as defined by the ARIA spec. What are you citing that's expecting this?

**Comment by chihch** [comment]
If the component has role="grid" (which is each section list in GroupedList) doesn't have the ariaLabel to its header text, screen reader will only reads out "table with 2 rows...." (assuming it's a list with two items). Screen reader user will not be able to know what table it is. In contrast, adding ariaLabel="it's parent header text" would make screen reader read out "<header text> table with 2 rows.....".

There is another way to fix it is that instead of adding ariaLabel, adding ariaLabelledby="<the id of it's parent header>". That way can achieve the same reading.

Is there any way we can override from our end, or we need to rely on Fluent team to fix it?

Thanks,
Eric

**Comment by chihch** [comment]
friendly reminder, any update on this? thanks.

**Comment by joschect** [comment]
@chihch I can see what you're talking about. It does seem like there should be an aria-labelled by on the grid content, at least according to the w3 examples with grid. 

For right now there isn't a way to override this.

@betrue-final-final or @cschlechty  might be able to comment more.

@chihch If you want to submit a fix that would help out a lot! It sounds like you've already done some research so if you submit a pr we can talk there further. Otherwise I've added this to our backlog.

**Comment by chihch** [comment]
awesome! thanks for confirming the bug. Yes, an aria-labelledBy on the <div role='grid'> is one option. Another option would be just use an aria-label=<section header text>. Either one should give screen reader the same reading i believe. I'll see what i can do for contributing cause our team is also trying to figure out the contribution model. thx @joschect and everyone

**Comment by chihch** [comment]
can we reopen this item? This is an accessibility issue. thanks so much.
